LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan/ExceptionFilterUtility.cs

```diff
@@ -7,15 +7,23 @@ namespace LoRaWan
 
     public static class ExceptionFilterUtility
     {
-        public static bool True(Action action)
+        public static bool True(params Action[] actions)
         {
-            (action ?? throw new ArgumentNullException(nameof(action)))();
+            if (actions is null) throw new ArgumentNullException(nameof(actions));
+
+            foreach (var a in actions)
+                a();
+
             return true;
         }
 
-        public static bool False(Action action)
+        public static bool False(params Action[] actions)
         {
-            (action ?? throw new ArgumentNullException(nameof(action)))();
+            if (actions is null) throw new ArgumentNullException(nameof(actions));
+
+            foreach (var a in actions)
+                a();
```

Comment:
If one of the 2 actions throws due to a programming mistake, should we make sure to execute the other one?

Reply:
While I do agree that it would be more safe, I think in this case (the `ExceptionFilterUtility` has a very distinct use - side effects in `when` clauses) the additional complexity from aggregating the exceptions etc is not worth it. We should make very sure that we don't mask the original exception by generating other exceptions in the `True` and `False` arguments.